lib/polaris/updates.ex

```diff
@@ -903,6 +903,66 @@ defmodule Polaris.Updates do
     {parent_init_fn, apply_fn}
   end
 
+  @doc """
+  Accumulate gradients and only apply updates every N steps.
+  """
+  def accumulate_gradients({parent_init_fn, parent_apply_fn} \\ identity(), steps)
+      when is_integer(steps) do
+    # We can't implement this one as stateful really because
+    # we don't want to run the updates at all nor change the optimizer
+    # state aside from this one
+    init_fn = fn params ->
+      state = parent_init_fn.(params)
+      Tuple.insert_at(state, 0, init_accumulate_gradients(params))
```

Comment:
I wonder if it would be best to do something like:
```suggestion
      {init_accumulate_gradients(params), state}
```

and then below we match on `{this_state, other_state}`

Reply:
I think the assumptions in the other updates depend on the tuple insertion here unfortunately. We could probably switch to an actual data structure for transformation states